mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+    """
+    A class for detecting and logging MLFlow runs into MLRun. Notice, only the last active MLFlow run is logged.
+    """
+
+    @classmethod
+    def is_enabled(cls) -> bool:
+        """
+        Checks if tracker is enabled.
+
+        :return: True if the tracking configuration is enabled, False otherwise.
+        """
+        return getattr(mlconf.external_platform_tracking, mlflow.__name__).enabled
+
+    @classmethod
+    def pre_run(cls, context: MLClientCtx):
+        """
+        Initializes the MLFlow tracker, setting the experiment name as configured by the user.
+
+        :param context: Current mlrun context
+        """
+        # Check for a user set experiment name via the environment variable:
+        experiment_name = mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.get()
+
+        # Check if the user configured for matching the experiment name with the context name:
+        if getattr(
+            mlconf.external_platform_tracking, mlflow.__name__
+        ).match_experiment_to_runtime:
+            if experiment_name is not None:
+                context.logger.warn(
+                    f"`mlconf.external_platform_tracking.mlflow.match_experiment_to_runtime` is set to True but the "
+                    f"MLFlow experiment name environment variable ('MLFLOW_EXPERIMENT_NAME') is set for using the "
+                    f"name: '{experiment_name}'. This name will be overriden "
+                    f"with MLRun's runtime name as set in the MLRun configuration: '{context.name}'."
+                )
+            experiment_name = context.name
+            mlflow.set_experiment(experiment_name=experiment_name)
+
+    @classmethod
+    def post_run(cls, context: MLClientCtx):
+        """
+        Performs post-run tasks for logging MLFlow artifacts generated during the run.
+
+        :param context: Current mlrun context
+        """
+        # Get the last run (None means no run was created or found):
+        run = mlflow.last_active_run()
+        if run is None:
+            return
+
+        # Log the run:
+        MLFlowTracker._log_run(context=context, run=run)
+
+    @classmethod
+    def import_run(
+        cls,
+        project: MlrunProject,
+        pointer: str,
+        function_name: str = None,
+        handler: str = None,
+    ) -> RunObject:
+        """
+        Import a previous MLFlow experiment run to MLRun.
+
+        :param project:       The MLRun project to import the run to.
+        :param pointer:       The MLFlow `run_id` to import.
+        :param function_name: The MLRun function to assign this run to.
+        :param handler:       The handler for MLRun's RunObject
+
+        :return: The newly imported RunObject.
+        """
+        # Validate function name was given:
+        if function_name is None:
+            raise ValueError(
+                "For importing a MLFlow experiment run, a MLRun function name must be provided."
+            )
```

Comment:
so, perhaps, do not default `function_name` to be None but force the callee to mention the function name?

Reply:
agreed

Reply:
The reason for it is this function is inheriting from the abstract one, so adding variables to it must have default values. In this case, I guess we can move the function name and handler to the main function as they are sort of mandatory for every tracking.